export_dagmc_cmd/DAGMCExportCommand.cpp

```diff
@@ -0,0 +1,825 @@
+#include "DAGMCExportCommand.hpp"
+#include "CubitInterface.hpp"
+
+// CGM includes
+#include "GeometryQueryTool.hpp"
+#include "ModelQueryEngine.hpp"
+#include "GMem.hpp"
+
+#include "RefEntityName.hpp"
+
+#include "Body.hpp"
+#include "Surface.hpp"
+#include "Curve.hpp"
+
+#include "RefGroup.hpp"
+#include "RefFace.hpp"
+#include "RefEdge.hpp"
+#include "RefVertex.hpp"
+
+#include "SenseEntity.hpp"
+
+// MOAB includes[
+#include "MBTagConventions.hpp"
+#include "moab/Core.hpp"
+#include "moab/Interface.hpp"
+#include "moab/GeomTopoTool.hpp"
+
+#define CHK_MB_ERR_RET(A,B)  if (moab::MB_SUCCESS != (B)) { \
+  message << (A) << (B) << std::endl;                                   \
+  CubitInterface::get_cubit_message_handler()->print_message(message.str().c_str()); \
+  return false;                                                         \
+  }
+
+#define CHK_MB_ERR_RET_MB(A,B)  if (moab::MB_SUCCESS != (B)) { \
+  message << (A) << (B) << std::endl;                                   \
+  return rval;                                                         \
+  }
+
+DAGMCExportCommand::DAGMCExportCommand() :
+  geom_tag(0), id_tag(0), name_tag(0), category_tag(0), faceting_tol_tag(0), geometry_resabs_tag(0)
+{
+  // set default values
+  norm_tol = 5;
+  faceting_tol = 1e-3;
+  len_tol = 0.0;
+  verbose_warnings = false;
+  fatal_on_curves = false;
+}
+
+DAGMCExportCommand::~DAGMCExportCommand()
+{}
+
+std::vector<std::string> DAGMCExportCommand::get_syntax()
+{
+  // Define the syntax for the command. Note the syntax is a modified BNF
+  // format. Full documentation on the command specification syntax can be
+  // found in the documentation.
+  std::string syntax =
+      "export dagmc "
+      "<string:label='filename',help='<filename>'> "
+      "[faceting_tolerance <value:label='faceting_tolerance',help='<faceting tolerance>'>] "
+      "[length_tolerance <value:label='length_tolerance',help='<length tolerance>'>] "
+      "[normal_tolerance <value:label='normal_tolerance',help='<normal tolerance>'>] "
+      //"[overwrite] "
```

Comment:
Definitely something we should include later - hopefully using existing utilities within Trelis, but for now this comment can go.